content/docs/user-guide/experiment-management/hydra-composition.md

```diff
@@ -221,4 +222,102 @@ Stage 'train' didn't change, skipping
+
+## Migrating Hydra Projects
+
+If you already have Hydra configured and want to start using DVC alongside it,
+you may need to refactor your code slightly. DVC will not pass the Hydra config
+to `@hydra.main()`, so it should be dropped from the code. Instead, DVC composes
+the Hydra config before your code runs and dumps the results to `params.yaml`.
+
+Using the example above, here's how the Python code in `train.py` might look
+using Hydra without DVC:
+
+```python
+import hydra
+from omegaconf import DictConfig
+
+@hydra.main(version_base=None, config_path="conf", config_name="config")
+def main(cfg: DictConfig) -> None:
+    # train model using cfg parameters
+
+if __name__ == "__main__":
+    main()
+```
+
+To convert the same code to use DVC with Hydra composition enabled:
+
+```python
+from omegaconf import OmegaConf
+
+def main() -> None:
+    cfg = OmegaConf.load("params.yaml")
+    # train model using cfg parameters
+
+if __name__ == "__main__":
+    main()
+```
+
+You no longer need to import Hydra into your code. A `main()` method is included
+in this example because it is good practice, but it's not necessary. This
+separation between config and code can help debug because the entire config
+generated by Hydra gets written to `params.yaml` before the experiment starts.
+You can also reuse `params.yaml` across multiple scripts in different stages of
+a DVC pipeline.
+
+## Advanced Hydra config
+
+You can configure how DVC works with Hydra.
+
+By default, DVC will look for Hydra [config groups] in a `conf` directory, but
+you can set a different directory using `dvc config hydra.config_dir other_dir`.
+This is equivalent to the `config_path` argument in `@hydra.main()`.
+
+Within that directory, DVC will look for [defaults list] in `config.yaml`, but
+you can set a different path using `dvc config hydra.config_name other.yaml`.
+This is equivalent to the `config_name` argument in `@hydra.main()`.
+
+Hydra will automatically discover plugins in the `hydra_plugins` directory. By
+default, DVC will look for `hydra_plugins` in the root directory of the DVC
+repository, but you can set a different path with
+`dvc config hydra.plugins_path other_path`.
+
+### Custom resolvers
+
+You can register [OmegaConf custom resolvers] as plugins by writing them to a
+file inside `hydra_plugins`. DVC will use these custom resolvers when composing
+the Hydra config. For example, add a custom resolver to
+`hydra_plugins/my_resolver.py`:
+
+```python
+import os
+from omegaconf import OmegaConf
+
+OmegaConf.register_new_resolver('join', lambda x, y : os.path.join(x, y))
+```
+
+You can use that custom resolver inside the Hydra config:
+
+```yaml
+dir: raw/data
+relpath: dataset.csv
+fullpath: ${join:${dir},${relpath}}
+```
+
+The final `params.yaml` will look like:
+
+```yaml
+dir: raw/data
+relpath: dataset.csv
+fullpath: raw/data/dataset.csv
+```
+
+[plugins]:
+  https://hydra.cc/docs/advanced/plugins/develop/#automatic-plugin-discovery-process
```

Comment:
I'm probably wrong, but you don't seem to use this link anywhere in the doc. Is that normal?

Reply:
Yes, good catch! I added a link above to reference this.